### ISSUE TITLE
Exponential smoothing back-fill BG gaps (dangerous)

### ISSUE THREAD

**Issue by justmara** [nightscout/AndroidAPS#2883]
Today I were changing sensor for my kid and got bad calibration from first attemp, so there was a noticeable gap between last and new BG values. First screenshot shows how (original) Tsunami smoothing that I've shown in that [PR](https://github.com/nightscout/AndroidAPS/pull/2141) handled that on 3.1.0.3 and second one shows how the same gap were handled in 3.2-dev-m with exponential smoothing on my test phone.

As you can see original algorithm simply ignores missing values, while exponential smoothing fills it with non-existing values. That could possibly lead to very dangerous behaviour.

![Screenshot_20231008_153506_NSClient](https://github.com/nightscout/AndroidAPS/assets/4157587/f04b9fd0-326d-4d09-aa8b-c2fbcc15440d)

![Screenshot_20231008_153518_AAPS](https://github.com/nightscout/AndroidAPS/assets/4157587/bab8d565-0538-48ba-8f6b-a462a4ed0099)

**Comment by ga-zelle** [comment]
Agreed; glucose data gaps should never be filled be it linearly interpolated or anything else. The fact of the gap is an important bit of information that should be kept in mind. For instance, SMB should be disabled for 10-15 minutes until a new trend is clear.

**Comment by ga-zelle** [comment]
That gap filling also happens without using any smoothing option inside AAPS3.2rc1

**Comment by MilosKozak** [comment]
filling the gap was always there. in older versions this was not displayed
without interpolation COB calculation would not be working

**Comment by TranceCake** [comment]
Isn't it good practise to disable the loop if you don't trust the accuracy of the BG anyway?

**Comment by justmara** [comment]
Completed? You've made really scary behaviour and only because noone ever knew It existed before - it is considered _safe_? Wtf?
AAPS must NOT auto-fill gaps with fake BG values. Thats unacceptable.
It can allign existing values to 5-min intervals with interpolation for calculation purpose, but NOT add non-existing values.
Here is an example how BG fall IS interpreted as raise because this fake backfill.

**Comment by MilosKozak** [comment]
this behavior is there the since 1st version of AAPS so i'd mark it time proved
and it's there only for calculation purposes, no BG is added

**Comment by blaqone** [comment]
> this behavior is there the since 1st version of AAPS so i'd mark it time proved
> and it's there only for calculation purposes, no BG is added

This is not true - it’s there since exp smoothing. And there seem to be a few bad implementations we NEED to talk about!!

**Comment by sennierer** [comment]
I realized that in the second graph there seem to be some interpolated data points (around 11) that are real data points in the first graph. Is that intended behavior @MilosKozak ? I realized that in my case (G7, xdrip+ and AAPS 3.2.0.2+autoISF3.0) AAPS seems to replace some of the xdrip values with interpolated ones (at least they appear in the light green color) and its also changing on every loop run which data points are exchanged for interpolated ones.

**Comment by MilosKozak** [comment]
@sennierer screenshot?

**Comment by sennierer** [comment]
@MilosKozak here are screenshots within the 36h period with the yellow (recalculated data) flag and screenshots after. After the the yellow flag is gone I dont have interpolated data, before I have. xdrip didnt miss any data in both cases.
![Screenshot_20231219-103000](https://github.com/nightscout/AndroidAPS/assets/499265/69036bef-35fe-46d9-bad2-d2a95c85eab5)
![Screenshot_20231219-102950](https://github.com/nightscout/AndroidAPS/assets/499265/8ef0d1f1-78fd-4681-a0d2-b7a85e07909b)
![Screenshot_20231218-115250](https://github.com/nightscout/AndroidAPS/assets/499265/6aa60ce3-57b8-436b-86ac-80af0e02ded3)
![Screenshot_20231218-115239](https://github.com/nightscout/AndroidAPS/assets/499265/9c4a7cc0-33c5-4261-8189-28cc183eb595)
![Screenshot_20231217-131027](https://github.com/nightscout/AndroidAPS/assets/499265/dc38a9b7-e764-4b8f-970e-7244a7cb6f42)

**Comment by sennierer** [comment]
and as you can see in the last screenshot delta, delta 15 min and acc are clearly wrong (not sure though if the 0 shown is an actual 0 or rather None or invalid datawise).
